design-proposals/mass_machine_type_transition.md

```diff
@@ -171,59 +225,66 @@ As both the minimum supported machine type version and the latest machine type v
+					* VM machine type is less than the minimum supported machine type
+					* VM machine type is greater than or equal to the minimum supported machine type
 					* VM machine type is equal to `q35`
 
 # Implementation Phases
-
- - [ ] Create MMTT package that Kubernetes job will use as an image
- - [ ] Create Kubernetes job yaml to be invoked with MMTT subcommand
- - [ ] Insert package and job into kubevirt/pkg/virtctl and implement the subcommand (name TBD)
+## Initial Phase
+This will be a bare-bones design that will be functional for the user, but with limited features that focus on the specific use case of upgrading the machine types of VMs with outdated or deprecated machine types to the latest supported machine type. The user will have the ability to use the command with the flags to specify by namespace, label-selector,  or to restart any running VMs immediately. In this stage, the minimum supported machine type and the latest machine type will be determined and stored internally; the user will not be able to configure what machine types they would like to use at this time. Additionally, to monitor and manage/delete the job, the user can utilize the `kubectl` commands. In future phases, we will add subcommands that allow the user to monitor and manage the specific machine type transition job directly.
+ - [ ] Create mass machine type transition package that k8s job will use as an image, and build the image.
+ - [ ] Add `virtctl` subcommand that will create the k8s job and run it, with flags for specifying by namespace, label-selector, and immediately restarting running VMs that have been affected.
+ - [ ] Add unit and functional tests.
+## Future Phases
+As follow-ups, we will implement the following features:
+ - Dynamically retrieving the supported and deprecated QEMU machine types and allowing the user to select a specific machine type to migrate to. Some more discussion and planning will be required to flesh out exactly how to implement this; namely how the user will know what machine types are supported and any restrictions we want to place on the user when selecting a machine type to convert to.
+ - Subcommands that the user can use to monitor and manage the jobs. These could include `virtctl convert-machine-types delete` which will safely terminate and clean up the job and delete it, and `virtctl convert-machine-types list` which will allow the user to see the status of the VMs being updated by the job.
+ - Other flags that the user can specify VMs to be affected by. For example, if the user wants to only convert a single specific VM, we might want to implement a flag to allow the user to specify the VM by name.
+ - When using the `restart-now` flag, allow the user to configure how many VMs they want to restart at once; depending on the number of running VMs, it may be very intensive to trigger the restart of possibly thousands of VMs at once.
```

Comment:
I'd suggest that once this proposal merges you'd open an issue to track these future steps